### PR TITLE
UAF-42 UAF-1834 Added the Conflict of Interest drop down list to the vendor document

### PR DESCRIPTION
**Pull request by leahasullivan**
Created edu/arizona version of ojb-vnd.xml to add the mapping for the VendorDetailExtension table and add the reference for the "extension" to the Vendor Detail table.  Updated the spring-vnd.xml with the database file path to the newly created ojb-vnd.xml file.  Created the VendorParameterConstants.java file to hold the parameter name of the parameter that holds the values for the drop down list.  Created the ConflictOfInterestTypeValuesFinder.java file to retrieve the values for the drop down list.  Created the VendorDetailExtension bo file to tie to the ojb mappings.  Created the VendorDetailExtension datadictionary file to define the bean for the Conflict of interest and point to the values finder class.  Updated the VendorMaintenanceDocument.xml file with the new field.  Updated the VendorDetail.xml file with the new conflict of interest field attribute definition and added it to the inquiry.  Table and parameter already exist.

### CODE REVIEW

**Comment by leahasullivan** [comment]
Fixed merge conflicts between spring-vnd.xml, ojb-vnd.xml, VendorDetailExtension.xml, VendorDetailExtension.java, and VendorDetail.xml.  Conflicts were between this mod and the mod for the eCustoms step 1 mod where both were editing the same files as both are extended attributes.

**Comment by leahasullivan** [comment]
Created the edu.arizona version of the VendorServiceImpl which implement the vendorService.  Added the methods necessary for the implementation.  Added methods shouldVendorRouteForApproval and noRouteSignificantChangeOccurredForExtensions.  This will compare the extended attributes to see if changes have occurred and if so, it will route the document for approval.

Added the bean for vendorService to point to the new VendorServiceImpl class, but reference the org.kuali version as well.

Added method isEqualForRouting to the VendorDetailExtension.java.

File: kfs-core/src/main/java/edu/arizona/kfs/vnd/document/service/impl/VendorServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+            throw new RuntimeException("A WorkflowException was thrown which prevented the loading of " + "the comparison document (" + documentId + ")", we);
+        }
+
+        if (ObjectUtils.isNotNull(newDoc)) {
+
+            VendorDetailExtension oldVDtlExtension =  (VendorDetailExtension) ((VendorDetail) newDoc.getOldMaintainableObject().getBusinessObject()).getExtension();
+            VendorDetailExtension newVDtlExtension = (VendorDetailExtension) ((VendorDetail)newDoc.getNewMaintainableObject().getBusinessObject()).getExtension();
+
+            if (ObjectUtils.isNotNull(oldVDtlExtension)) {
+                shouldRoute = !(noRouteSignificantChangeOccurredForExtensions(oldVDtlExtension, newVDtlExtension));
```
Comment (ghost) [inline kfs-core/src/main/java/edu/arizona/kfs/vnd/document/service/impl/VendorServiceImpl.java:134]: Call to method noRouteSignificantChangeOccurredForExtensions has oldVDtlExtension is first in parameter list. In noRouteSignificantChangeOccurredForExtensions method declaration, newVDtlExtension is first in parameter list.

File: kfs-core/src/main/java/edu/arizona/kfs/vnd/document/service/impl/VendorServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+    public VendorService getVendorService() {
+        return vendorService;
+    }
+
+    public void setVendorService(VendorService vendorService) {
+        this.vendorService = vendorService;
+    }
+
+    public DocumentService getDocumentService() {
```
Comment (ghost) [inline kfs-core/src/main/java/edu/arizona/kfs/vnd/document/service/impl/VendorServiceImpl.java]: Service getters are not needed.

File: kfs-core/src/main/java/edu/arizona/kfs/vnd/document/service/impl/VendorServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+        boolean unchanged = oldVDtlExtension.isEqualForRouting(newVDtlExtension);
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Exiting noRouteSignificantChangeOccurredForExtensions.");
+        }
+        
+        return unchanged;
+    }
+
+    public VendorService getVendorService() {
```
Comment (ghost) [inline kfs-core/src/main/java/edu/arizona/kfs/vnd/document/service/impl/VendorServiceImpl.java]: Service getters are not needed.

**Comment by ghost** [comment]
Completed initial review. Please see my comments.

**Comment by leahasullivan** [comment]
Removed getter methods for vendorService and documentService. Changed the orders of the arguments for method noRouteSignificatChangeOccurredForExtensions to match the method call.  Changed the call to isEqualForRouting method to reflect the change in the method signature.

**Comment by ghost** [comment]
Looks good!

**Pull request merged**